File: spring-orm/src/main/java/com/beneboba/spring_orm/Blog.java
package com.beneboba.spring_orm;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.jpa.repository.JpaRepository;

@Data
@NoArgsConstructor
@Entity
@Table(name = "BLOG_DETAILS")
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "category")
    private String category;

    @Column(name = "content")
    private String content;

    @JsonIgnore
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "owner_id")
    private Owner owner;

    public Blog(String title, String category, String content) {
        this.title = title;
        this.category = category;
        this.content = content;
    }
}

File: spring-orm-kotlin/src/main/kotlin/com/beneboba/spring_orm_kotlin/entity/Blog.kt
package com.beneboba.spring_orm.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.Data
import lombok.NoArgsConstructor

@Data
@NoArgsConstructor
@Entity
@Table(name = "BLOG_DETAILS")
class Blog(
    @field:Column(name = "title") private val title: String,
    @field:Column(name = "category") private val category: String,
    @field:Column(
        name = "content"
    ) private val content: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private val id: Long? = null

    @JsonIgnore
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "owner_id")
    private val owner: Owner? = null
}